File: Java/Easy/2507-number-of-common-factors/number-of-common-factors.java
class Solution {
    public int commonFactors(int a, int b) {
        int i,count=0,limit;

        if (a<b)
        limit=b;
        else
        limit=a;
        
        for(i=1;i<=limit;i++){
        if(a%i==0 && b%i==0){
            count++;
        }
     }
        return count;
    }
}

File: C_Plus_Plus/Easy/2507-number-of-common-factors/number-of-common-factors.cpp
class Solution {
public:
    int commonFactors(int a, int b) {
        int count=0;
        int c=max(a,b); // To find the ending range
        for(int i=1;i<=c;i++){
            if(a%i==0 && b%i==0)
            count++;
        }
        return count;
    }
};